packages/frontend/src/utils/fetch.ts

```diff
@@ -6,7 +6,13 @@ import type { Payload } from '@hapi/boom';
 
 type Routes = AMARoutes & AuthRoutes & ModmailRoutes;
 
-export class APIError extends Error {
+export class APIFetchError extends Error {
+	public constructor(public readonly payload: Payload) {
+		super(payload.message);
+	}
+}
+
+export class APIMutateError extends Error {
```

Comment:
I think I'd rather we kept this one class and added a prop to indicate the method that was used.